### ISSUE TITLE
[Python3] NameError when running salt-run manage.versions

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#42697]
### Description of Issue/Question

```
[root@ch3ll-master-py3 ~]# salt-run manage.versions

Exception occurred in runner manage.versions: Traceback (most recent call last):
  File "/root/salt/salt/client/mixins.py", line 392, in _low
    data['return'] = self.functions[fun](*args, **kwargs)
  File "/root/salt/salt/runners/manage.py", line 673, in versions
    ver_diff = cmp(minion_version, master_version)
NameError: name 'cmp' is not defined
```

Seems cmp() was removed from python3.

### Setup
install salt with python3
### Steps to Reproduce Issue
`salt-run manage.versions`

**Comment by Ch3LL** [comment]
@rallytime assigned to you for the 2017.7.2 release.

**Comment by rallytime** [comment]
@Ch3LL This should be fixed with #42782.
